src/statistical/unigram.jl

```diff
@@ -0,0 +1,179 @@
+"""
+struct Sentencepiecemodel
+  vocab::Array{String,1}
+  logprob::Array{Float64,1}
+end
+structure, To hold vocabulary,log probability and index  
+"""
+struct Sentencepiecemodel
+  vocab::Array{String,1}
+  logprob::Array{Float64,1}
+end
+
+function load(ty::Type{T}, name::String) where T<:PretrainedTokenizer
+    filepath = @datadep_str name
+    filepath = "$filepath/$name"
+    load(filepath)  
+end
+    
+function load(path)
+    vocab = readlines(path)
+    vocabnew = split.(vocab , "\t")
+    vo = []
+    for i in 1:length(vocab)
+        vocab1 = vocabnew[i][1]
+        vocab1 = replace(vocab1,"▁"=>"_")
+        push!(vo,vocab1)
+    end
+    vocab1 = convert(Array{String,1},vo)
+    logprob = []
+    for i in 1:length(vocab)
+        logp = vocabnew[i][2]
+        push!(logprob,logp)    
+    end
+    logp = convert(Array{String,1},logprob)
+    logp =parse.(Float64,logprob)
+    spm = Sentencepiecemodel(vocab1,logp)
+    return spm
+end
+
+# to get index of given string
+function getindex(sp::Sentencepiecemodel,text)
+    id_list = findall(x->x==text, sp.vocab)
+    if length(id_list) == 0
+        return 2
+    end
```

Comment:
```suggestion
    length(id_list) == 0 && return 2
```
?

Reply:
Why return 2, here, is it the index for Out-of-vocab-Tokens?

Reply:
Unk token